base/applications/osk/main.c

```diff
@@ -204,6 +237,9 @@ int OSK_DlgClose(void)
     /* Save the settings to the registry hive */
     SaveDataToRegistry();
 
+    /* Destroy the dialog and its operations */
+    EndDialog(Globals.hMainWnd, FALSE);
```

Comment:
This was not needed before?
Why is it needed now?

Reply:
Because when you click on "Exit", the dialog won't close and you're forced to click on "X" to do that. Weirdly this happens only if the dialog has a menu.